From: Nick Rosbrook <enr0n@ubuntu.com>
Date: Wed, 23 Jul 2025 09:37:47 -0400
Subject: test: follow /usr/bin/sleep symlink when checking coredumps

With the coreutils-from-gnu and coreutils-from-uutils logic, coreutils
binaries are symlinked depending on the coreutils-from-* package
installed. E.g., we may have /usr/bin/sleep -> /usr/bin/gnusleep.

This causes TEST-87-AUX-UTILS-VM to fail thinking there are no matching
coredumps.

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/2116465
---
 test/units/TEST-87-AUX-UTILS-VM.coredump.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/units/TEST-87-AUX-UTILS-VM.coredump.sh b/test/units/TEST-87-AUX-UTILS-VM.coredump.sh
index 19f0fa2..5d830b8 100755
--- a/test/units/TEST-87-AUX-UTILS-VM.coredump.sh
+++ b/test/units/TEST-87-AUX-UTILS-VM.coredump.sh
@@ -97,12 +97,12 @@ EOF
     machinectl start "$CONTAINER"
     timeout "$TIMEOUT" bash -xec "until systemd-run -M '$CONTAINER' -q --wait --pipe true; do sleep .5; done"
 
-    [[ "$(systemd-run -M "$CONTAINER" -q --wait --pipe coredumpctl list -q --no-legend sleep | wc -l)" -eq 0 ]]
+    [[ "$(systemd-run -M "$CONTAINER" -q --wait --pipe coredumpctl list -q --no-legend $(readlink -f /usr/bin/sleep) | wc -l)" -eq 0 ]]
     machinectl copy-to "$CONTAINER" "$MAKE_DUMP_SCRIPT"
     systemd-run -M "$CONTAINER" -q --wait --pipe "$MAKE_DUMP_SCRIPT" "sleep" "SIGABRT"
     systemd-run -M "$CONTAINER" -q --wait --pipe "$MAKE_DUMP_SCRIPT" "sleep" "SIGTRAP"
     # Wait a bit for the coredumps to get processed
-    timeout 30 bash -c "while [[ \$(systemd-run -M $CONTAINER -q --wait --pipe coredumpctl list -q --no-legend sleep | wc -l) -lt 2 ]]; do sleep 1; done"
+    timeout 30 bash -c "while [[ \$(systemd-run -M $CONTAINER -q --wait --pipe coredumpctl list -q --no-legend $(readlink -f /usr/bin/sleep) | wc -l) -lt 2 ]]; do sleep 1; done"
 
     machinectl stop "$CONTAINER"
     rm -rf "/var/lib/machines/$CONTAINER"
